ssh-krb-node-executor-plugin/contents/scp-krb-executor.sh

```diff
@@ -13,36 +13,49 @@ shift
 SRC_FILE=$1
 shift
 DEST_FILE=$1
-KERB_KEYTAB=$RD_CONFIG_KERBEROS_KEYTAB
 
-if [ ! -f "$KERB_KEYTAB" ]; then
-  echo "Keytab $KERB_KEYTAB not found" >&2
+KRB_USER=$RD_CONFIG_KERBEROS_USER
+KRB_KEYTAB=$RD_CONFIG_KERBEROS_KEYTAB
+USE_CUSTOM_KRB_CACHE=$RD_CONFIG_USE_KERBEROS_CUSTOM_CACHE_FILE
+KRB_CACHE_FILENAME=$RD_CONFIG_KERBEROS_CUSTOM_CACHE_FILENAME
+KRB_DESTROY=$RD_CONFIG_DO_KDESTROY
+
+
+if [ ! -f "$KRB_KEYTAB" ]; then
+  echo "Keytab $KRB_KEYTAB not found" >&2
   exit 2
 fi
 
+
 # random delay (0..0.5s) to make it work with parallel exec
 sleep "$(bc <<< "scale=2; $(printf '0.%02d' $(( RANDOM % 100))) / 2")"
 
-# status 1 if the credentials cache cannot be read or is expired, and with status 0 otherwise
-klist -s
-if [ $? -ne 0 ]; then
-  kinit -kt "$KERB_KEYTAB" "$RD_CONFIG_KERBEROS_USER"
 
-  # verify ticket has been successfuly created
-  klist -s
-  if [ $? -ne 0 ]; then
-    echo "Kinit failure when calling kinit -kt $KERB_KEYTAB $RD_CONFIG_KERBEROS_USER" >&2
+# override the default kerberos cache file if required
+if [[ "$USE_CUSTOM_KRB_CACHE" == 'true' ]]; then
+  if [ ! -z "$KRB_CACHE_FILENAME" ]; then
```

Comment:
What about this?
```suggestion
  if [ -n "$KRB_CACHE_FILENAME" ]; then
```

Shellcheck also suggests this change.

Reply:
[done](https://github.com/cernops/rundeck-exec-kerberos/pull/7/commits/26601ace8e46b9883c82bffe721d88e8fb22eb4a)